File: test90_out/local.mcu/ConsolePrint/ConsolePrint.js

/*
 * ======== MODULE local.mcu.ConsolePrint ========
 */

var $$c = function() {
    const ConsolePrint = {};
    ConsolePrint.ConsolePrint = ConsolePrint
    ConsolePrint.$name = 'local.mcu.ConsolePrint';
    ConsolePrint.pollen$used = 0;
    ConsolePrint.printBool = new $$Ref('local_mcu_ConsolePrint_printBool__E');
    ConsolePrint.printInt = new $$Ref('local_mcu_ConsolePrint_printInt__E');
    ConsolePrint.printReal = new $$Ref('local_mcu_ConsolePrint_printReal__E');
    ConsolePrint.printUint = new $$Ref('local_mcu_ConsolePrint_printUint__E');
    ConsolePrint.printStr = new $$Ref('local_mcu_ConsolePrint_printStr__E');
    ConsolePrint.targetInit = new $$Ref('local_mcu_ConsolePrint_targetInit__I');
    ConsolePrint.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    ConsolePrint.pollen__uses__ = function() {
    }
    ConsolePrint.$$privateInit = function() {
        ConsolePrint.HEADER = "\033[95m";
        ConsolePrint.OKBLUE = "\033[94m";
        ConsolePrint.OKGREEN = "\033[92m";
        ConsolePrint.WARNING = "\033[93m";
        ConsolePrint.FAIL = "\033[91m";
        ConsolePrint.ENDC = "\033[0m";
    }
    return ConsolePrint;
}
var $$u = $$c();
$units['local.mcu.ConsolePrint'] = $$u;
$units.push($$u);

File: test90_out/local.mcu/ConsolePrint/ConsolePrint.c

/*
 * ======== string literals ========
 */

static const char local_mcu_ConsolePrint__s__fileName[] = "ConsolePrint.p";
static const char local_mcu_ConsolePrint__s__2[] = "\033[92m";
static const char local_mcu_ConsolePrint__s__4[] = "\033[91m";
static const char local_mcu_ConsolePrint__s__0[] = "\033[95m";
static const char local_mcu_ConsolePrint__s__1[] = "\033[94m";
static const char local_mcu_ConsolePrint__s__3[] = "\033[93m";
static const char local_mcu_ConsolePrint__s__5[] = "\033[0m";
static const char local_mcu_ConsolePrint__s__6[] = "ConsolePrint";

/*
 * ======== code injections ========
 */

 #include <stdio.h> 

/*
 * ======== function definitions ========
 */

void local_mcu_ConsolePrint_printUint__E( uint32 u ) {
     printf("%iu", u) ;

}

void local_mcu_ConsolePrint_printInt__E( int32 i ) {
     printf("%i", i) ;

}

void local_mcu_ConsolePrint_printReal__E( float f ) {
     printf("%f", (float)f) ;

}

void local_mcu_ConsolePrint_printBool__E( bool b ) {
    if ((b)) {
         printf("true") ;

    }
    else {
         printf("false") ;

    }
}

void local_mcu_ConsolePrint_targetInit__I() {
}

void local_mcu_ConsolePrint_printStr__E( string s ) {
     printf("%s", s) ;

}

